### ISSUE TITLE
-Wsometimes-uninitialized in drivers/bluetooth/btmtksdio.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#421]
```
drivers/bluetooth/btmtksdio.c:392:6: warning: variable 'old_len' is used uninitialized whenever 'if' condition is true [-Wsometimes-uninitialized]
        if (err < 0)
            ^~~~~~~
drivers/bluetooth/btmtksdio.c:471:20: note: uninitialized use occurs here
                       old_data, old_len, true);
                                 ^~~~~~~
drivers/bluetooth/btmtksdio.c:392:2: note: remove the 'if' if its condition is always false
        if (err < 0)
        ^~~~~~~~~~~~
drivers/bluetooth/btmtksdio.c:376:22: note: initialize the variable 'old_len' to silence this warning
        unsigned int old_len;
                            ^
                             = 0
1 warning generated.
```

**Comment by nathanchance** [comment]
Patch submitted by the author of the offending code: https://lore.kernel.org/lkml/741bf172b7d0b77e664e7edeab1bd798c955cc5b.1552942432.git.sean.wang@mediatek.com/

**Comment by nathanchance** [comment]
Fixed in `next-20190320`: https://git.kernel.org/next/linux-next/c/5488c82e67f278c6519396fb8d2b3baf2932b312
